Refactor projects and update CMakeLists.txt.

Craig-Stoneham opened the pull request:
Refactor projects into "Games" directory and renamed them so there is a bit more consistency. Bump CMake to version 3.8 so we have access to some of the later functionality. Remove CMakeLists.txt from projects. We now only have one CMake file in the repo root. Inlined Input::listeners temporarily as I was getting an error (LNK2001: unresolved external symbol) in Visual Studio.

Craig-Stoneham committed:
Refactor projects and update CMakeLists.txt.

Refactor projects into "Games" directory and renamed them so there is a bit more consistency.
Bump CMake to version 3.8 so we have access to some of the later functionality.
Remove CMakeLists.txt from projects. We now only have one CMake file in the repo root.
Inlined Input::listeners temporarily as I was getting an error (LNK2001: unresolved external symbol) in Visual Studio.

ural89 reviewed (changes requested):
that works great. Only we need to add ncurses library here to make it work in linux
   # Link with Engine library
    target_link_libraries(${PROJECT} PRIVATE Engine ncurses)

Craig-Stoneham committed:
Update CMakeLists.txt

Added list of platforms with platform option.
Corrected start up project as it was initially set to an invalid option.

Craig-Stoneham committed:
Update .gitignore

Added ignore file directories:
bin
compiler

ural89 committed:
renaming conio

Craig-Stoneham committed:
Refactor codebase and remove compile warnings.

Refactored code into a more logical structure.
Rename conio.h to Console.h and EntryPoint.h into Engine.h.
Scatter 'static_cast' around the codebase, where applicable, to remove compilation warnings.
Tidied up the source code  in places.

Craig-Stoneham committed:
Merge branch 'main' of https://github.com/Craig-Stoneham/ConsoleCraftEngine

Craig-Stoneham committed:
Update CMakeLists.txt

Added apply_common_settings for all builds.

Craig-Stoneham committed:
Add common pre processor macros

Craig-Stoneham committed:
Update Console.h

Update file to include our platform pre processor macros.

Craig-Stoneham committed:
Update CMakeLists.txt

Make CMAKE_INSTALL_PREFIX and CMAKE_CONFIGURATION_TYPES options INTERNAL.

Craig-Stoneham committed:
Update CMakeLists.txt

Add CMAKE_WINDOWS_EXPORT_ALL_SYMBOLS so that VS build tools also exports a library with a DLL.

ural89 committed:
build fix for linux
